### PR TITLE
INS-216: Move common ledger types to core package.

### PR DESCRIPTION
**Pull request by andreyromancev**
Closes INS-216

**- What I did**
Moved common ledger types to core package.

**- How to verify it**
Run tests

### CODE REVIEW

File: core/artifactmanager.go [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+	// AppendObjDelegate creates append object record in storage. Provided reference should be a reference to the head
+	// of the object. Provided memory well be used as append delegate memory.
+	//
+	// Object's delegates will be provided by GetLatestObj. Any object update will nullify all the object's append
+	// delegates. VM is responsible for collecting all appends and adding them to the new memory manually if its
+	// required.
+	AppendObjDelegate(domain, request, obj RecordRef, memory []byte) (RecordRef, error)
+}
+
+type ClassDescriptor interface {
```
Comment (nordicdyno) [inline core/artifactmanager.go]: godoc

File: core/artifactmanager.go [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+	// GetCode fetches the latest class code known to storage. Code will be fetched according to architecture preferences
+	// set via SetArchPref in artifact manager. If preferences are not provided, an error will be returned.
+	GetCode() ([]byte, error)
+
+	// GetMigrations fetches all migrations from provided to artifact manager state to the last state known to storage. VM
+	// is responsible for applying these migrations and updating objects.
+	GetMigrations() ([][]byte, error)
+}
+
+type ObjectDescriptor interface {
```
Comment (nordicdyno) [inline core/artifactmanager.go]: godoc

File: core/record.go [hunk 3]
```diff
@@ -0,0 +1,4 @@
+package core
```
Comment (nordicdyno) [inline core/record.go]: license

Comment (ita-sammann) [reply core/record.go]: ... and doc.go file.

**Pull request merged**